libs/api/domains/air-discount-scheme/src/lib/api-domains-air-discount-scheme.resolver.ts

```diff
@@ -0,0 +1,61 @@
+import { UseGuards } from '@nestjs/common'
+import { Query } from '@nestjs/graphql'
+import type { User } from '@island.is/auth-nest-tools'
+import { ApiScope } from '@island.is/auth/scopes'
+import {
+  CurrentUser,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import { Audit } from '@island.is/nest/audit'
+import { AirDiscountSchemeService } from './api-domains-air-discount-scheme.service'
+
+import {
+  Discount as TDiscount,
+  User as TUser,
+} from '@island.is/air-discount-scheme/types'
+import { Discount } from '../models/discount.model'
+
+type DiscountWithTUser = Discount & { user: TUser }
+
+const TWO_HOURS = 7200
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.airDiscountScheme)
```

Comment:
I imagine this will never support custom delegations, so we should use internal scopes instead of creating a new one (still use token exchange).

What's the plan for children coupons? Is it going to be like today where you can get them while logged in as the parent. Or are we going to use delegation system so you switch to your kid first to get their coupons?

Reply:
1) I assume you mean changing to `@island.is/internal` on `ApiScope.internal`. Yeah, that makes sense. My original thinking was to have this minimally exposed (very explicit) but that may be undue paranoia.
I'll get on that :)

2) Users have come to expect seeing themselves and their children in one screen but I can see how delegations should be standardized moving forward. When I had this discussion with "Hugsmiðjan" we discussed changing the user experience as little as possible. It shouldn't be difficult to change especially since the system was written with that in mind that children might log in to begin with.

Reply:
Oh although, about the internal scope. Is there a problem regarding guardians (is: forsjá) in that one? 
I am not too well versed in the IDS setup but looking in https://identity-server.dev01.devland.is/admin/resource/api-scope/%40island.is%2Finternal we see that the Delegation-> Grant To Legal Guardians is unchecked.

I assume this is fine since we are exchanging to a token with this enabled?

Reply:
1. Yes. The internal scope is designed for exactly this use case, to avoid exploding our tokens with too many scopes.
2. Either is fine. It just affects which scopes you need. If the parent sees everything and should not see this menu for their children, then @island.is/internal is enough. If we want the parent to see this when they switch to their children, we should also create/add @island.is/internal:forsja.

There's shouldn't be any problem with guardians and the internal scope, unless we want the user to be able to see coupons for their children using the delegation system.

Reply:
Switched out for internal scope. Works on local. Will test on feature deploy.